sdl-core/src/test/scala/io/smartdatalake/workflow/action/ScalaClassSparkDs2To1TransformerTest.scala

```diff
@@ -0,0 +1,227 @@
+      .withColumn("concatenated_name", concat($"ds1.name", $"ds2.name"))
+      .select("concatenated_name", "added_rating")
+      .as[AnotherOutputDataSet]
+    result
+  }
+}
+
+class ScalaClassSparkDs2To1TransformerTest extends FunSuite with BeforeAndAfter {
+
+  protected implicit val session: SparkSession = TestUtil.sessionHiveCatalog
+
+  import sessionHiveCatalog.implicits._
+
+  private val tempDir = Files.createTempDirectory("testScalaClassSparkDs2To1TransformerTest")
+  private val tempPath = tempDir.toAbsolutePath.toString
+
+  implicit val instanceRegistry: InstanceRegistry = new InstanceRegistry
+  implicit val contextExec: ActionPipelineContext = TestUtil.getDefaultActionPipelineContext.copy(phase = ExecutionPhase.Exec)
+
+  before {
+    instanceRegistry.clear()
+  }
+
+  test("One DS2To1 Transformation (direct call to exec)") {
+    // setup DataObjects
+    // source has partition columns dt and type
+    val srcDO1 = CsvFileDataObject("src1DS", tempPath + "/src1", partitions = Seq("name")
+      , schema = Some(SparkSchema(StructType.fromDDL("name string, rating int"))))
+    instanceRegistry.register(srcDO1)
+    val srcDO2 = CsvFileDataObject("src2DS", tempPath + "/src2", partitions = Seq("name")
+      , schema = Some(SparkSchema(StructType.fromDDL("name string, rating int"))))
+    instanceRegistry.register(srcDO2)
+    val tgt1DO = CsvFileDataObject("tgt1DS", tempPath + "/tgt1", partitions = Seq()
+      , schema = Some(SparkSchema(StructType.fromDDL("concatenated_name string, added_rating int"))))
+    instanceRegistry.register(tgt1DO)
+
+    // fill src with first files
+    val dfSrc1 = Seq(("john", 5))
+      .toDF("name", "rating")
+    srcDO1.writeSparkDataFrame(dfSrc1, Seq())
+    val dfSrc2 = Seq(("doe", 10))
+      .toDF("name", "rating")
+    srcDO2.writeSparkDataFrame(dfSrc2, Seq())
+
+
+    // prepare & start load
+    val customTransformerConfig = ScalaClassSparkDs2To1Transformer(className = classOf[TestResolutionByIdDS2To1Transformer].getName)
+    val testAction = CustomDataFrameAction("action", List(srcDO1.id, srcDO2.id), List(tgt1DO.id), transformers = Seq(customTransformerConfig))
+
+    instanceRegistry.register(srcDO1)
+    instanceRegistry.register(srcDO2)
+    instanceRegistry.register(tgt1DO)
+    instanceRegistry.register(testAction)
+
+    val srcSubFeed1 = SparkSubFeed(None, "src1DS", partitionValues = Seq())
+    val srcSubFeed2 = SparkSubFeed(None, "src2DS", partitionValues = Seq())
+    testAction.exec(Seq(srcSubFeed1, srcSubFeed2))
+
+    val actual = tgt1DO.getSparkDataFrame().as[AnotherOutputDataSet].head()
+    assert(actual.added_rating == 15)
+    assert(actual.concatenated_name == "johndoe")
+  }
+
+  test("One DS2To1 Transformation using config file: two identical input types using dataObjectId") {
+
+    // setup DataObjects
+    // source has partition columns dt and type
+    val srcDO1 = CsvFileDataObject("src1", "target/src1DS2to1", partitions = Seq("name")
```

Comment:
Can we use a tempPath here as well?
I suggest to use a different tempPath per unit test.

Reply:
I don't know how to synchronize the dynamically generated location of the temp path with the location of the dataObjects defined in the config files. Maybe using environment variables would be an option, but I don't think this is a good idea. What do you think?
I'm writing to target because I got inspired by io.smartdatalake.app.SmartDataLakeBuilderTest ;)